client/context.go

```diff
@@ -212,27 +212,30 @@ func (ctx Context) PrintString(str string) error {
 // either text or json. If text, toPrint will be YAML encoded. Otherwise, toPrint
 // will be JSON encoded using ctx.JSONMarshaler. An error is returned upon failure.
 func (ctx Context) PrintOutput(toPrint proto.Message) error {
-	return ctx.printOutput(toPrint)
+	// always serialize JSON initially because proto json can't be directly YAML encoded
+	out, err := ctx.JSONMarshaler.MarshalJSON(toPrint)
+	if err != nil {
+		return err
+	}
+	return ctx.printOutput(out)
```

Comment:
So we actually don't want to call `printOutput` here because that uses amino. Instead, we want to take care of amino in `PrintOutputLegacy` and just have `printOutput` to string printing.